### PR TITLE
[RPS-426] Add upcoming facet

### PR DESCRIPTION
**Pull request by jacsamell**
Also [RPS-428] - Move date facets to top

Added upcoming facet to search results and included upcoming
publications in search by setting the searchable flag.

### CODE REVIEW

File: acceptance-tests/src/test/resources/features/site/facets.feature [hunk 1]
```diff
@@ -134,3 +131,11 @@ Feature: Faceted search
...
         And I should see the list with title "GEOGRAPHICAL GRANULARITY" including:
             | Cancer networks   |
+
+    Scenario: Upcoming publications are returned
+        Given I navigate to the "search" page
+        Then I should see the list with title "PUBLICATION STATUS" containing:
+            | Published ( ... |
+            | Upcoming (2) |
+        When I click on the "Upcoming" button
+        Then I can click on the "Upcoming Publication" link
```
Comment (wojtek-oledzki) [inline acceptance-tests/src/test/resources/features/site/facets.feature]: simples :)

File: repository-data/application/src/main/resources/hcm-config/configuration/update/Migration20180319UpcomingPublicationsSearchable.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class Migration20180319UpcomingPublicationsSearchable extends BaseNodeUpdateVisitor {
+
+    private static final String PROPERTY_SEARCHABLE = "common:searchable"
+
+    boolean doUpdate(Node node) {
+        try {
+            log.info("attempting to migrate node: " + node.getPath())
+
+            JcrUtils.ensureIsCheckedOut(node)
+            node.setProperty(PROPERTY_SEARCHABLE, true)
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-config/configuration/update/Migration20180319UpcomingPublicationsSearchable.groovy]: is it worth having a more generic script - "Set Searchable Flag" - with a parameter to set it to `true` or `false`?

File: repository-data/webfiles/src/main/resources/site/freemarker/common/facets.ftl [hunk 3]
```diff
@@ -32,15 +32,20 @@
...
+                    <#assign valueName="Not Defined"/>
                     <#if facet.name="month">
                         <@fmt.message key=value.name var="monthName"/>
                         <#assign valueName=monthName/>
                     <#elseif facet.name="category">
                         <#assign valueName=taxonomy.getValueName(value.name)/>
                     <#elseif facet.name="document-type">
-                        <@fmt.message key="facet."+value.name var="valueName"/>  
+                        <@fmt.message key="facet."+value.name var="docType"/>
+                        <#assign valueName=docType/>
```
Comment (wojtek-oledzki) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/facets.ftl]: I didn't get it :( Why 2 assignments?

**Comment by wojtek-oledzki** [comment]
@jacsamell do you have plan for applying new facet config? (and labels before deployment)

File: repository-data/application/src/main/resources/hcm-config/configuration/update/Migration20180319UpcomingPublicationsSearchable.groovy (hunk 2 above)
Comment (jacsamell) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/Migration20180319UpcomingPublicationsSearchable.groovy]: As discussed don't think it's worth it in this case

File: repository-data/webfiles/src/main/resources/site/freemarker/common/facets.ftl (hunk 3 above)
Comment (jacsamell) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/facets.ftl]: For some reason the `<@fmt.message key="facet."+value.name var="docType"/>` variable appears to be somewhat final in that you can't reassign it in the second facet type. Where as if you call `#assign` you can reassign the variable. Without this the second use of `@fmt.message` doesn't work and you just get the previous value repeated. Weird!

**Pull request merged**